fastly/helpers.go

```diff
@@ -0,0 +1,27 @@
+package fastly
+
+import "strings"
+
+// escapeSpecialChars uses Terraform's escape syntax (i.e., repeating characters)
+// to properly escape symbols (e.g., '%', '$', etc).
+//
+// There are two significant places where symbols can show up:
+// 1. Before a left curly brace (i.e., '{').
+// 2. Not before a left curly brace.
+//
+// In case #1, we have to double escape so that Terraform does not interpret Fastly's
+// configuration values as its own (e.g., https://docs.fastly.com/en/guides/custom-log-formats).
+//
+// In case #2, we only have to single escape.
+//
+// Refer to the Terraform documentation on string literals for more details:
+// https://www.terraform.io/docs/configuration/expressions.html#string-literals
+func escapeSpecialChars(s string) string {
+	escapePercent := strings.ReplaceAll(s, "%", "%%")
+	escapePercentTemplate := strings.ReplaceAll(escapePercent, "%%{", "%%%%{")
+
+	escapeDollarSign := strings.ReplaceAll(escapePercentTemplate, "$", "$$")
```

Comment:
It can't be this naive. Diff when trying to use the helper in DataDog tests.

```
        -               `  "is_cacheable": %{if(fastly_info.state~"^(HIT|MISS)$", "true","false")}V,`,
        +               `  "is_cacheable": %{if(fastly_info.state~"^(HIT|MISS)$$", "true","false")}V,`,
```

Reply:
@phamann Maybe for now just escape '%'? Or omit the test helper?